lib/cosmos/packet_logs/packet_log_reader.rb

```diff
@@ -227,58 +160,94 @@ def read(identify_and_define = true)
       raise err
     end
 
+    # TODO: Currently not used
+    # Returns an analysis of the log file by reading all the packets and
+    # returning information about each packet. This information maps directly
+    # to the parameters need by the {#read_at_offset} method and thus should be
+    # called before using {#read_at_offset}.
+    #
+    # @param filename [String] The filename to analyze
+    # @param progress_callback [Proc] Callback that should receive a single
+    #   floating point parameter which is the percentage done
+    # @return [Array<Array<Integer, Integer, String, String, Time, Time>] Array
+    #   of arrays for each packet found in the log file consisting of:
+    #   [File position, length, target name, packet name, time formatted,
+    #   received time].
+    # def packet_offsets(filename, progress_callback = nil)
+    #   open(filename)
+    #   offsets = []
+    #   filesize = size().to_f
+
+    #   while true
+    #     current_pos = @file.pos
+    #     packet = read(false)
+    #     break unless packet
+    #     offsets << current_pos
+    #     if progress_callback
+    #       break if progress_callback.call(current_pos / filesize)
+    #     end
+    #   end
+
+    #   return offsets
+    # ensure
+    #   close()
+    # end
+
+    # TODO: Currently not used
     # Reads a packet from the opened log file. Should only be used in
     # conjunction with {#packet_offsets}.
     #
     # @param file_offset [Integer] Byte offset into the log file to start
     #   reading
     # @param identify_and_define (see #each)
     # @return [Packet]
-    def read_at_offset(file_offset, identify_and_define = true)
-      @file.seek(file_offset, IO::SEEK_SET)
-      return read(identify_and_define)
-    rescue => err
-      close()
-      raise err
-    end
-
+    # def read_at_offset(file_offset, identify_and_define = true)
+    #   @file.seek(file_offset, IO::SEEK_SET)
+    #   return read(identify_and_define)
+    # rescue => err
+    #   close()
+    #   raise err
+    # end
+
+    # TODO: Currently not used
     # Read the first packet from the log file and reset the file position back
     # to the current position. This allows the client to call read multiple
     # times to return packets, call first, and continue calling read which will
     # return the next packet in the file.
     #
     # @return [Packet]
-    def first
-      original_position = @file.pos
-      @file.seek(0, IO::SEEK_SET)
-      read_file_header()
-      packet = read()
-      raise "No first packet found" unless packet
-      @file.seek(original_position, IO::SEEK_SET)
-      packet.clone
-    rescue => err
-      close()
-      raise err
-    end
-
+    # def first
```

Comment:
I commented out the unused methods for now so we're clear they haven't been tested (also they aren't used anywhere yet)